File: PHP/Film.php
<?php
class Film {
    private int $id;
    private string $judul;
    private string $genre;
    private string $jamTayang;
    private string $gambar;

    // Constructor
    public function __construct(int $id, string $judul, string $genre, string $jamTayang, string $gambar) {
        $this->id = $id;
        $this->judul = $judul;
        $this->genre = $genre;
        $this->jamTayang = $jamTayang;
        $this->gambar = $gambar;
    }

    // Getter
    public function getId(): int { return $this->id; }
    public function getJudul(): string { return $this->judul; }
    public function getGenre(): string { return $this->genre; }
    public function getJamTayang(): string { return $this->jamTayang; }
    public function getGambar(): string { return $this->gambar; }

    // Setter
    public function setJudul(string $judul): void { $this->judul = $judul; }
    public function setGenre(string $genre): void { $this->genre = $genre; }
    public function setJamTayang(string $jamTayang): void { $this->jamTayang = $jamTayang; }
    public function setGambar(string $gambar): void { $this->gambar = $gambar; }
}

File: CPP/Film.cpp
#include "Film.h"
#include <iostream>
using namespace std;

Film::Film() {}

Film::Film(int id, string judul, string genre, string jamTayang, string gambar) {
    this->id = id;
    this->judul = judul;
    this->genre = genre;
    this->jamTayang = jamTayang;
    this->gambar = gambar;
}

int Film::getId() { return id; }
string Film::getJudul() { return judul; }
string Film::getGenre() { return genre; }
string Film::getJamTayang() { return jamTayang; }
string Film::getGambar() { return gambar; }

void Film::setJudul(string judul) { this->judul = judul; }
void Film::setGenre(string genre) { this->genre = genre; }
void Film::setJamTayang(string jamTayang) { this->jamTayang = jamTayang; }
void Film::setGambar(string gambar) { this->gambar = gambar; }

void Film::tampilkan() {
    cout << "ID           : " << id << endl;
    cout << "Judul        : " << judul << endl;
    cout << "Genre        : " << genre << endl;
    cout << "Jam Tayang   : " << jamTayang << endl;
    cout << "Gambar       : " << gambar << endl;
    cout << "-----------------------------" << endl;
}

Film::~Film() {}
